cpp/CMakePresets.json

```diff
@@ -1,165 +1,231 @@
+        "ARROW_BUILD_TESTS": "ON",
+        "ARROW_COMPUTE": "ON",
+        "ARROW_CSV": "ON",
+        "ARROW_DATASET": "ON",
+        "ARROW_EXTRA_ERROR_CONTEXT": "ON",
+        "ARROW_FILESYSTEM": "ON",
+        "ARROW_JSON": "ON"
+      }
+    },
+    {
+      "name": "ninja-debug",
+      "inherits": "ninja-debug-basic",
+      "displayName": "Debug build with tests and more optional components",
+      "cacheVariables": {
+        "ARROW_ENGINE": "ON",
+        "ARROW_MIMALLOC": "ON",
+        "ARROW_PARQUET": "ON",
+        "ARROW_WITH_BROTLI": "ON",
+        "ARROW_WITH_BZ2": "ON",
+        "ARROW_WITH_LZ4": "ON",
+        "ARROW_WITH_RE2": "ON",
+        "ARROW_WITH_SNAPPY": "ON",
+        "ARROW_WITH_UTF8PROC": "ON",
+        "ARROW_WITH_ZLIB": "ON",
+        "ARROW_WITH_ZSTD": "ON"
+      }
+    },
+    {
+      "name": "ninja-debug-cuda",
+      "inherits": "ninja-debug-basic",
+      "displayName": "Debug build with tests and CUDA integration",
+      "cacheVariables": {
+        "ARROW_CUDA": "ON"
+      }
+    },
+    {
+      "name": "ninja-debug-filesystems",
+      "inherits": "ninja-debug-basic",
+      "displayName": "Debug build with tests and filesystems",
+      "cacheVariables": {
+        "ARROW_GCS": "ON",
+        "ARROW_HDFS": "ON",
+        "ARROW_S3": "ON"
+      }
+    },
+    {
+      "name": "ninja-debug-flight",
+      "inherits": "ninja-debug-basic",
+      "displayName": "Debug build with tests and Flight",
+      "cacheVariables": {
+        "ARROW_FLIGHT": "ON"
+      }
+    },
+    {
+      "name": "ninja-debug-gandiva",
+      "inherits": "ninja-debug-basic",
+      "displayName": "Debug build with tests and Gandiva",
+      "cacheVariables": {
+        "ARROW_GANDIVA": "ON"
+      }
+    },
+    {
+      "name": "ninja-debug-python",
+      "inherits": "ninja-debug-basic",
+      "displayName": "Debug build with tests and Python support",
+      "cacheVariables": {
+        "ARROW_PYTHON": "ON"
+      }
+    },
+    {
+      "name": "ninja-debug-maximal",
+      "inherits": "ninja-debug",
+      "displayName": "Debug build with everything enabled (except benchmarks and CUDA)",
+      "cacheVariables": {
+        "ARROW_BUILD_EXAMPLES": "ON",
+        "ARROW_BUILD_UTILITIES": "ON",
+        "ARROW_FLIGHT": "ON",
+        "ARROW_GANDIVA": "ON",
+        "ARROW_GCS": "ON",
+        "ARROW_HDFS": "ON",
+        "ARROW_HIVESERVER2": "ON",
+        "ARROW_ORC": "ON",
+        "ARROW_PLASMA": "ON",
+        "ARROW_PYTHON": "ON",
+        "ARROW_S3": "ON",
+        "ARROW_SKYHOOK": "ON",
+        "ARROW_TENSORFLOW": "ON",
+        "PARQUET_BUILD_EXAMPLES": "ON",
+        "PARQUET_BUILD_EXECUTABLES": "ON",
+        "PARQUET_REQUIRE_ENCRYPTION": "ON"
+      }
+    },
+
+    {
+      "name": "ninja-release-minimal",
+      "inherits": "ninja-debug-minimal",
+      "displayName": "Release build without anything enabled",
+      "cacheVariables": {
+        "ARROW_BUILD_INTEGRATION": "OFF",
```

Comment:
This is already specified in parent unless the parent has it set to `OFF` incorrectly.